AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/ProcessMutationErrorFromCloudOperation.swift

```diff
@@ -33,68 +43,306 @@ class ProcessMutationErrorFromCloudOperation: Operation {
         log.verbose(#function)
 
         guard !isCancelled else {
-            queryOperation?.cancel()
-            let apiError = DataStoreError.unknown("Operation cancelled", "")
-            finish(result: .failure(apiError))
+            let error = DataStoreError.unknown("Operation cancelled", "")
+            finish(result: .failure(error))
             return
         }
 
-        processConditionalRequestFailed()
-    }
+        guard case let .error(graphQLErrors) = error else {
+            finish(result: .success(nil))
+            return
+        }
 
-    private func processConditionalRequestFailed() {
-        if case let .error(graphQLErrors) = error {
-            // TODO: Check for 'ConflictUnhandled', execute conflict handler
+        guard graphQLErrors.count == 1 else {
+            finish(result: .success(nil))
+            return
+        }
```

Comment:
i was going back and forth on this as well, to be future proof, it would make sense to simply process all errors in the array, however i don't think this operation should be responsible for that (it would also make the unit tests more complicated). perhaps another operation or method which enqueues each operation with a single error. currently i didn't have an example where there would be an additional error in the array so i didn't have a meaningful test to write. I think an improvement is to log it here so that we know what the additional errors are and then we can revisit the scenario and account for it.